src/1D/spline_cache_1D.jl

```diff
@@ -264,7 +264,7 @@ function CubicBSpline(x::Vector, y::Vector; end_condition = "free", smoothing_fa
          1  4  1 0]
 
   # Free end condition
-  if end_condition == "free"
+  if end_condition === "free"
```

Comment:
```suggestion
  if end_condition == "free"
```
Only symbols should be compared using `===` instead of `==`, strings should use the usual comparison.

Reply:
Ah, whoops. I got this backwards. I thought that `===` would be better because of strings. I will change it back.